packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -209,131 +248,143 @@ class MouseTracker extends ChangeNotifier {
   /// MouseTracker. Do not call in other contexts.
   @visibleForTesting
   bool isAnnotationAttached(MouseTrackerAnnotation annotation) {
-    return _trackedAnnotations.containsKey(annotation);
+    return _trackedAnnotations.contains(annotation);
   }
 
-  /// Tells interested objects that a mouse has entered, exited, or moved, given
-  /// a callback to fetch the [MouseTrackerAnnotation] associated with a global
-  /// offset.
+  /// Collect the latest states of the given mouse devices, and call interested
+  /// callbacks.
+  ///
+  /// Only those devices of [deviceIds] are updated. If [deviceIds] is null,
+  /// then all devices are updated. For each of device, the enter or exit events
+  /// are called for annotations that the pointer enters or leaves, while hover
+  /// events are always called for each annotations that the pointer stays in,
+  /// even if the pointer has not moved since the last call. Therefore it's
+  /// caller's responsibility to check if the pointer has moved.
   ///
-  /// This is called from a post-frame callback when the layer tree has been
-  /// updated, right after rendering the frame.
+  /// This is called synchronously in most cases, except for when a new
+  /// annotation is attached, which is called in a post frame callback.
   ///
   /// This function is only public to allow for proper testing of the
   /// MouseTracker. Do not call in other contexts.
   @visibleForTesting
   void sendMouseNotifications(Iterable<int> deviceIds) {
```

Comment:
Yes, it is the plan. It's not done in this PR because changing the signature is breaking. I have a PR (https://github.com/flutter/flutter/pull/41879) that makes this method private, after which we can do whatever change that makes it pretty.